tests/Oryx.Integration.Tests/DotNetCore/DotNetCorePreRunCommandOrScriptTest.cs

```diff
@@ -0,0 +1,137 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Oryx.BuildScriptGeneratorCli;
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.Tests.Common;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Oryx.Integration.Tests
+{
+    [Trait("category", "dotnetcore")]
+    public class DotNetCorePreRunCommandOrScriptTest : DotNetCoreEndToEndTestsBase
+    {
+        public DotNetCorePreRunCommandOrScriptTest(ITestOutputHelper output, TestTempDirTestFixture testTempDirTestFixture)
+            : base(output, testTempDirTestFixture)
+        {
+        }
+        [Theory]
+        [InlineData("2.1", NetCoreApp21WebApp, "Hello World!")]
+        [InlineData("3.1", NetCoreApp31MvcApp, "Welcome to ASP.NET Core MVC!")]
+        public async Task CanBuildAndRun_NetCore31WebApp_UsingPreRunCommand_WithDynamicInstalls(
+            string runtimeVersion,
+            string appName,
+            string expectedResponseContent)
+        {
+            // Arrange
+            var hostDir = Path.Combine(_hostSamplesDir, "DotNetCore", appName);
+            var volume = DockerVolume.CreateMirror(hostDir);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = $"{appDir}/myoutputdir";
+            var buildImageScript = new ShellScriptBuilder()
+               .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+               .AddCommand(
+                $"oryx build {appDir} --platform dotnet --platform-version {runtimeVersion} -o {appOutputDir}")
+               .ToString();
+            var runtimeImageScript = new ShellScriptBuilder()
+                .SetEnvironmentVariable(SettingsKeys.EnableDynamicInstall, true.ToString())
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .SetEnvironmentVariable(FilePaths.PreRunCommandOrScriptEnvVarName, "echo \"Here has a pre run command.\"")
```

Comment:
How are you testing that this command ran successfully? One way to achieve it here is to create a file in the `appOutputDir` and assert this file's existence at line 77